ClosedXML/Excel/Tables/XLTable.cs

```diff
@@ -860,5 +865,369 @@ internal IXLTable CopyTo(XLWorksheet targetSheet, bool copyData = true)
             }
             return newTable;
         }
+
+        #region Append and replace data
+
+        public IXLRange AppendData(IEnumerable data)
+        {
+            return AppendData(data, false);
+        }
+
+        public IXLRange AppendData(IEnumerable data, bool transpose)
+        {
+            if (data == null || data is String)
+                return null;
+
+            var numberOfNewRows = data.Cast<object>().Count();
+            var lastRowOfOldRange = this.DataRange.LastRow();
+            lastRowOfOldRange.InsertRowsBelow(numberOfNewRows);
+
+            return OverwriteTableData(lastRowOfOldRange.RowBelow(), data, transpose);
+        }
+
+        public IXLRange AppendData(DataTable dataTable)
+        {
+            return AppendData(dataTable.Rows.Cast<DataRow>());
+        }
+
+        public IXLRange AppendData<T>(IEnumerable<T> data)
+        {
+            if (!data.Any() || data is String)
+                return null;
+
+            var numberOfNewRows = data.Count();
+            var lastRowOfOldRange = this.DataRange.LastRow();
+            lastRowOfOldRange.InsertRowsBelow(numberOfNewRows);
+
+            return OverwriteTableData(lastRowOfOldRange.RowBelow(), data);
+        }
+
+        public IXLRange ReplaceData(IEnumerable data)
+        {
+            return ReplaceData(data, false);
+        }
+
+        public IXLRange ReplaceData(IEnumerable data, bool transpose)
+        {
+            if (data == null || data is String)
+                throw new InvalidOperationException("Cannot replace table data with empty enumerable.");
+
+            // Resize table
+            var sizeDifference = data.Cast<object>().Count() - this.DataRange.RowCount();
+            if (sizeDifference > 0)
+                this.DataRange.LastRow().InsertRowsBelow(sizeDifference);
+            else if (sizeDifference < 0)
+                this.DataRange.Rows(this.LastRow().RowNumber() + sizeDifference + 1, this.LastRow().RowNumber())
+                    .OrderByDescending(r => r.RowNumber())
+                    .ForEach(r => r.Delete());
+
+            this.DataRange.Clear(XLClearOptions.Contents);
+
+            return this.OverwriteTableData(this.DataRange.FirstRow(), data, transpose);
+        }
+
+        public IXLRange ReplaceData(DataTable dataTable)
+        {
+            return ReplaceData(dataTable.Rows.Cast<DataRow>());
+        }
+
+        public IXLRange ReplaceData<T>(IEnumerable<T> data)
+        {
+            if (!data.Any() || data is String)
+                throw new InvalidOperationException("Cannot replace table data with empty enumerable.");
```

Comment:
AFAIK it will create a corrupt file if you completely remove the data range. Even in Excel it's not possible to completely clear the table. It always leaves at least 1 blank row. Rather than deal with that complexity, I prohibited it... for now.

Reply:
yup, confirmed. Deleting the data range completely will corrupt the file.